src/main/scala/com/vngrs/etl/codecs/Codec.scala

```diff
@@ -0,0 +1,15 @@
+package com.vngrs.etl.codecs
+
+import com.vngrs.etl.configs.Configs
+
+/**
+  * Generic Converter Codec Interface
+  *
+  * @tparam A Type of the input
+  * @tparam B Type of the output
+  * @tparam CONF Type of the configurations
+  */
+trait Codec[A, B, CONF <: Configs] extends Serializable {
```

Comment:
This type class has several problems:

1. Storing config in type class is not a good idea. It doesn't need it. It should only know how to operate on one type.
2. method should have a more descriptive name. If it is a codec, there should be _decode_ and _encode_ methods.
3. Type class instances should be composable. That is how complex type class instances build from simple ones. We must have an instance for `Timestamp`s and `Row`'s instance should use it. However, passing it via config prevents that so it is an antipattern.